plugins/src/main/java/com/google/fhir/gateway/plugin/SmartFhirScope.java

```diff
@@ -21,72 +21,53 @@
-  }
+  private static final Pattern VALID_SCOPE_PATTERN =
+      Pattern.compile(
+          "(\\buser|patient|system\\b)(\\/((\\*)|([a-zA-Z]+)))(\\.((\\*)|([cruds]+)|(\\bread|write\\b)))");
+  static final String ALL_RESOURCE_TYPES_WILDCARD = "*";
+  private static final String ALL_RESOURCE_PERMISSIONS_WILDCARD = "*";
+  private static final String SMART_V1_READ_RESOURCE_PERMISSIONS = "read";
+  private static final String SMART_V1_WRITE_RESOURCE_PERMISSIONS = "write";
 
-  Set<SmartFhirScopeResourcePermission> resourcePermissions;
+  @Getter private final PrincipalContext principalContext;
+  @Getter private final String resourceType;
+  @Getter private final Set<Permission> permissions;
 
   private SmartFhirScope(
-      SmartFhirScopeResourcePrincipalContext principalContext,
-      String resourceType,
-      Set<SmartFhirScopeResourcePermission> resourcePermissions) {
+      PrincipalContext principalContext, String resourceType, Set<Permission> resourcePermissions) {
     this.principalContext = principalContext;
-    this.resourcePermissions = resourcePermissions;
+    this.permissions = resourcePermissions;
     this.resourceType = resourceType;
   }
 
-  private static final Pattern PATIENT_SCOPE_PATTERN =
-      Pattern.compile(
-          "(\\buser|patient|system\\b)(\\/((\\*)|([a-zA-Z]+)))(\\.((\\*)|([cruds]+)|(\\bread|write\\b)))");
-  static final String ALL_RESOURCE_TYPES_WILDCARD = "*";
-  static final String ALL_RESOURCE_PERMISSIONS_WILDCARD = "*";
-  static final String SMART_V1_READ_RESOURCE_PERMISSIONS = "read";
-  static final String SMART_V1_WRITE_RESOURCE_PERMISSIONS = "write";
-
-  private static final List<SmartFhirScopeResourcePermission> resourcePermissionsSequence =
-      List.of(
-          SmartFhirScopeResourcePermission.CREATE,
-          SmartFhirScopeResourcePermission.READ,
-          SmartFhirScopeResourcePermission.UPDATE,
-          SmartFhirScopeResourcePermission.DELETE,
-          SmartFhirScopeResourcePermission.SEARCH);
-
   static List<SmartFhirScope> extractSmartFhirScopesFromTokens(List<String> tokens) {
     List<SmartFhirScope> scopes = new ArrayList<>();
     for (String scope : tokens) {
-      if (PATIENT_SCOPE_PATTERN.matcher(scope).matches()) {
+      if (VALID_SCOPE_PATTERN.matcher(scope).matches()) {
         scopes.add(createSmartScope(scope));
       }
     }
     return scopes;
   }
 
   private static SmartFhirScope createSmartScope(String scope) {
-    String[] principalTokenSplit = scope.split("/");
-    SmartFhirScopeResourcePrincipalContext principalContext =
-        SmartFhirScopeResourcePrincipalContext.getPrincipalContext(principalTokenSplit[0]);
-    String[] resourceTypeAndPermissionSplit = principalTokenSplit[1].split("\\.");
-    if (!isValidResourceType(resourceTypeAndPermissionSplit[0])) {
+    String[] split = scope.split("/");
+    PrincipalContext principalContext = PrincipalContext.getPrincipalContext(split[0]);
+    String[] permissionSplit = split[1].split("\\.");
```

Comment:
nit: Should we do a range check on the `split` array size before accessing its second element? In general I think it is a good practice to document the expected conditions of arguments using the `Preconditions.check*` methods.